sources/models/src/lib.rs

```diff
@@ -516,8 +517,8 @@ struct OciDefaults {
 ///// The hard and soft limit values for an OCI defaults resource limit.
 #[model(add_option = false)]
 struct OciDefaultsResourceLimit {
-    hard_limit: u32,
-    soft_limit: u32,
+    hard_limit: OciDefaultsResourceLimitValue,
+    soft_limit: OciDefaultsResourceLimitValue,
```

Comment:
This will require a data migration for the "max open files" limit - on downgrade, we need to turn `OciDefaultsResourceLimitValue` back into a `u32`, and on upgrade we might need to go the other way.